### PR TITLE
Community dropdown

### PR DESCRIPTION
**Pull request by luqmaan**
Add the community list dropdown and add the route `/map/community/:id`.

image

image

#184

### CODE REVIEW

File: src/FloodsRoutes.js [hunk 1]
```diff
@@ -67,7 +67,10 @@ class FloodsRoutes extends Component {
           <Route component={PublicHeader} />
         </Switch>
 
-        <Route path="/map" component={CrossingMapPage} />
+        <Switch>
+          <Route exact path="/map/community/:selectedCommunityId" component={CrossingMapPage} />
```
Comment (briaguya) [inline src/FloodsRoutes.js]: Yes! Setting a good pattern for sharable links!

File: src/components/Shared/CrossingMapPage/CrossingMapPage.js [hunk 2]
```diff
@@ -44,10 +59,30 @@ class CrossingMapPage extends Component {
       selectedCommunity: null,
       viewport: viewportAndCenter.viewport,
       center: viewportAndCenter.center,
-      mapCenter: viewportAndCenter.center,
     };
   }
 
+  componentDidUpdate(prevProps) {
+    const didLoad = !getIsLoading(this.props) && getIsLoading(prevProps);
```
Comment (briaguya) [inline src/components/Shared/CrossingMapPage/CrossingMapPage.js]: nice refactor!

File: src/stories/crossingListItem.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import React from 'react';
 import { storiesOf } from '@storybook/react';
+import { MemoryRouter } from 'react-router';
```
Comment (briaguya) [inline src/stories/crossingListItem.js]: Hooray for good tests!

Comment (luqmaan) [reply src/stories/crossingListItem.js]: Haha too bad this doesn't really confirm that the tests are good or bad. 😉

**Pull request merged**